File: MagisterGL/MagisterSceneObjC/Model/3d/Color.m
#import "Color.h"

@implementation Color

- (instancetype)init
{
  self = [super init];
  if (self) {
    _red = 0.0;
    _green = 0.0;
    _blue = 0.0;
    _value = 0.0;
  }
  return self;
}

- (UIColor *)getColor
{
  return [UIColor colorWithRed:(CGFloat)(self.red/255.0)
                         green:(CGFloat)(self.green/255.0)
                          blue:(CGFloat)(self.blue/255.0)
                         alpha:1.0f];
}

- (SCNVector3)getColorVector
{
  return SCNVector3Make((float)(self.red/255.0),
                        (float)(self.green/255.0),
                        (float)(self.blue/255.0));
}

@end

File: MagisterGL/MagisterGL/Model/3d/Color.swift
import UIKit
import SceneKit

class Color: NSObject
{
   var value: Double = 0.0
   var red: Double = 0
   var green: Double = 0
   var blue: Double = 0
   
   public var color: UIColor
   {
      get
      {
         return UIColor(red: CGFloat(red / 255.0),
                        green:  CGFloat(green / 255.0),
                        blue: CGFloat(blue / 255.0),
                        alpha: 1.0)
      }
   }
   
   public var colorVector: SCNVector3
   {
      get
      {
          return SCNVector3Make(Float(red / 255.0), Float(green / 255.0), Float(blue / 255.0))
      }
   }
   
   override init()
   {
   }
}
